lib/ansible_tower_client/job_template.rb

```diff
@@ -10,12 +10,19 @@ def initialize(api, raw_body)
       super
     end
 
-    def launch(vars = {})
+    def launch(options = {})
       launch_url = "#{url}launch/"
-      extra = JSONValues.new(vars).extra_vars
-      resp = api.post(launch_url, extra).body
-      job = JSON.parse(resp)
-      api.jobs.find(job['job'])
+      new_limit = options.delete('limit')
+      resp = with_temporary_changes(new_limit) do
+        api.post(launch_url, options).body
+      end
+
+      begin
+        job = JSON.parse(resp)
+        api.jobs.find(job['job'])
+      rescue ResourceNotFound
+        Error.parse(resp)
+      end
```

Comment:
What is the return on success and failure? Is it either a `Job` or an error string? An error should be thrown in case of failure.